request.go

```diff
@@ -158,7 +158,7 @@ func (r *Request) Retry() error {
 
 // Do submits the request
 func (r *Request) Do() error {
-	return r.collector.scrape(r.URL.String(), r.Method, r.Depth, r.Body, r.Ctx, *r.Headers, !r.collector.AllowURLRevisit)
+	return r.collector.scrape(r.URL.String(), r.Method, r.Depth+1, r.Body, r.Ctx, *r.Headers, !r.collector.AllowURLRevisit)
```

Comment:
I don't understand this change. Could you please clarify? The test you added doesn't catch it if I revert this change.

Reply:
Odd, the `TestCollectorDepth` test should catch that, it's so that the depth increases on each request, so that you can use `MaxDepth` in Queue-based crawlers. I'll take a look at it.